src/components/select/filterable-select/filterable-select.stories.tsx

```diff
@@ -753,3 +753,65 @@ export const SelectionConfirmedStory = () => {
 };
 
 SelectionConfirmedStory.parameters = { chromatic: { disableSnapshot: true } };
+
+export const CustomFilterAndOptionStyle = () => {
+  const [filterText, setFilterText] = useState("");
+  const [selectedColor, setSelectedColor] = useState<string | undefined>();
+
+  const data = useMemo(
+    () =>
+      [
+        { text: "Amber", color: "#FFBF00" },
+        { text: "Black", color: "#000000" },
+        { text: "Blue", color: "#0000FF" },
+        { text: "Brown", color: "#A52A2A" },
+        { text: "Green", color: "#008000" },
+        { text: "Orange", color: "#FFA500" },
+        { text: "Pink", color: "#FFC0CB" },
+        { text: "Purple", color: "#800080" },
+        { text: "Red", color: "#FF0000" },
+        { text: "White", color: "#FFFFFF" },
+        { text: "Yellow", color: "#FFFF00" },
+      ].filter(
+        ({ text }) =>
+          !filterText || text.toLowerCase().includes(filterText.toLowerCase())
```

Comment:
Suggestion (non-blocking): it might be worth adding a `trim` to the `filterText` here to account for whitespace etc. It's not massively relevant to this example but likely will be in more detailed implementations so could be worth having in the code example etc

```
!filterText || (filterText.trim().length && text.toLowerCase().includes(filterText.trim().toLowerCase()))

Reply:
Thinking about this some more I'm happy with how it's implemented so lets just leave it and if we need to make a change later it's only docs etc